bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieImpl.java

```diff
@@ -931,6 +931,18 @@ private Journal getJournal(long ledgerId) {
         return journals.get(MathUtils.signSafeMod(ledgerId, journals.size()));
     }
 
+    @VisibleForTesting
+    public ByteBuf createMasterKeyEntry(long ledgerId, byte[] masterKey) {
+        // new handle, we should add the key to journal ensure we can rebuild
+        ByteBuffer bb = ByteBuffer.allocate(8 + 8 + 4 + masterKey.length);
```

Comment:
If we’re returning a ByteBuf and it’s properly ref counted, then we should just use a pooled ByteBuf instance instead of wrapping a ByteBuffer

Reply:
@merlimat Great understanding! Thanks for your review and suggestion. I have used a pooled ByteBuf instead of wrapping a ByteBuffer. :)